src/Splitio-net-core/Services/Client/Classes/SelfRefreshingClient.cs

```diff
@@ -272,6 +268,26 @@ private void BuildBlockUntilReadyService()
         {
             _blockUntilReadyService = new SelfRefreshingBlockUntilReadyService(gates, splitFetcher, selfRefreshingSegmentFetcher, treatmentLog, eventLog, _log);
         }
-        #endregion
+
+        private void LaunchTaskSchedulerOnReady()
+        {
+            Task workerTask = Task.Factory.StartNew(
+                () => {
+                    while (true)
+                    {
+                        if (gates.IsSDKReady(0))
+                        {
+                            selfRefreshingSegmentFetcher.StartScheduler();
+                            break;
+                        }
+#if NETSTANDARD
```

Comment:
What about using a wrapper between Task and ThreadUtils to call here and in `SelfUpdatingEventLog` the same Wrapper?